spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/AbsCriterionSpec.scala

```diff
@@ -20,14 +20,9 @@ import com.intel.analytics.bigdl.tensor.Tensor
 import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
 
 @com.intel.analytics.bigdl.tags.Serial
-class AbsCriterionSpec extends FlatSpec with BeforeAndAfter with Matchers{
-  before {
-    if (!TH.hasTorch()) {
-      cancel("Torch is not installed")
-    }
-  }
-
-  "A Abs Criterion " should "generate correct output and grad" in {
+class AbsCriterionSpec extends TorchSpec with BeforeAndAfter with Matchers{
+    "A Abs Criterion " should "generate correct output and grad" in {
+    torchCheck()
```

Comment:
Why not just check this in `TorchSpec` with a `beforeAll`? then you don't need this call at all.
`TorchSpec` can extend `BeforeAndAfter`, `BeforeAndAfterAll` and `Matchers` and then all these tests don't have to.

This is the change I was finishing locally and it works fine that way. It still generates an `*** ABORTED ***` message of course.